### ISSUE TITLE
Improve system architecture detection

### ISSUE THREAD

**Issue by Sharparam** [CoraleStudios/Colore#41]
The current method for detecting whether the host operating system is 32 or 64-bit is not 100% reliable. When re-compiling Colore for .NET 3.5 and running in Unity, it will claim the system is 32-bit when in fact it is 64-bit.

[This StackOverflow answer](http://stackoverflow.com/a/28866330/1104531) contains code closely mimicking that used by the .NET Framework itself for [`Environment.Is64BitOperatingSystem`](https://msdn.microsoft.com/en-us/library/system.environment.is64bitoperatingsystem%28v=vs.110%29.aspx).

As Colore is currently targeting .NET 4.5 we can use `Environment.Is64BitOperatingSystem` directly.

For discussion about re-targeting Colore against 3.5 (or lower?) for increased compatibility I will create another issue.

Update: Discussion about re-targeting in #42.

**Comment by njbmartin** [comment]
To clarify, the issue is actually within Unity itself and not because of the downgraded framework target.  I have raised it as a bug with Unity and will add details here as they respond.

I also raised it as a question on their Answers site: http://answers.unity3d.com/questions/1066853/unity-64-bit-getpekind-not-being-returned-correctl.html

I have confirmed at least that the Stackoverflow answer mentioned above reports 64-bit correctly on a 64-bit machine (even if the application is running in 32-bit mode). Worth testing using a 32-bit system.
